client/state/data-layer/wpcom/read/tags/utils.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { map, compact, concat, isObject, isArray } from 'lodash';
+import { decodeEntities } from 'lib/formatting';
+
+/**
+ * Normalize response from the api so whether we get back a single tag or a list of tags
+ * we always pass forward a list
+ * Also transform the api response to be something more calypso-friendly
+ *
+ * @param  {Tag|Tags} apiResponse api response from the tags endpoint
+ * @return {Tags} An object containing all of the normalized tags in the format:
+ *  [
+ *    { id, displayName, url, title, slug },
+ *    ...
+ *  ]
+ */
+export function fromApi( apiResponse ) {
+	if ( ! apiResponse || ( ! apiResponse.tag && ! apiResponse.tags ) ) {
+		if ( process.env.NODE_ENV === 'development' ) {
+			throw new Error( 'bad api response for /read/tags' );
+		}
+		return [];
+	}
+
+	const tags = compact( concat(
+		[],
+		isObject( apiResponse.tag ) && apiResponse.tag,
+		isArray( apiResponse.tags ) && apiResponse.tags,
+	) );
+
+	return map( tags, tag => ( {
+		id: tag.ID,
+		displayName: decodeEntities( tag.display_name ),
+		url: `/tag/${ tag.slug }`,
+		title: decodeEntities( tag.title ),
+		slug: tag.slug.toLowerCase(),
+	} ) );
+}
```

Comment:
I'm really ambivalent on this. Is it nice to document the data we're expecting? Yes. Do we need to ship those bytes in production....? I'm less convinced there. 

I'd be fine with validating the data shape in a guarded section, but I'm really meh about altering it to use camelCase over the normal API shape. It's CPU time, memory churn, and bytes on the wire we don't really need to ship.